cmd/topology/main.go

```diff
@@ -103,15 +108,45 @@ func main() {
 		}
 	}
 
+	var enableDebug bool
+	// DEBUG is only read as an environment variable
+	debugEnv := viper.GetString("DEBUG")
+	if debugEnv != "" {
+		var err error
+		if enableDebug, err = strconv.ParseBool(debugEnv); err != nil {
+			logger.WithError(err).WithField("debug", debugEnv).Fatal("debug value is invalid")
+		}
+	}
+
 	svc := &service.Service{
 		VolumeFinder: volumeFinder,
 		CertFile:     certFile,
 		KeyFile:      keyFile,
 		Port:         bindPort,
 		Logger:       logger,
+		EnableDebug:  enableDebug,
 	}
 
 	if err := entrypoint.Run(context.Background(), svc); err != nil {
 		logger.WithError(err).Fatal("running service")
 	}
 }
+
+func updateTracing(logger *logrus.Logger) {
+	zipkinURI := viper.GetString("ZIPKIN_URI")
+	zipkinServiceName := viper.GetString("ZIPKIN_SERVICE_NAME")
+	zipkinProbability := viper.GetFloat64("ZIPKIN_PROBABILITY")
+
+	tp, err := tracer.InitTracing(zipkinURI, zipkinServiceName, zipkinProbability)
+	if err != nil {
+		logger.WithError(err).Error("initializing tracer")
+	}
+
+	if tp != nil {
```

Comment:
Nil check shouldn't be necessary if the previous err was non-nil.

Reply:
fixed